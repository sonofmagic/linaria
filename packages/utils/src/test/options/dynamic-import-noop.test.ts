import { parse } from '@babel/core';
import generate from '@babel/generator';
import traverse from '@babel/traverse';
import * as t from '@babel/types';

import dynamic from '../../options/dynamic-import-noop';

describe('dynamic-import-noop', () => {
  it('case 0', () => {
    const result = parse(`import('aaa')`, {
      filename: __filename,
    });
    // const s = dynamic({ types });
    traverse(result, {
      Import(path) {
        const noop = t.arrowFunctionExpression([], t.identifier('undefined'));

        path.parentPath.replaceWith(
          t.objectExpression([
            t.objectProperty(t.identifier('then'), noop),
            t.objectProperty(t.identifier('catch'), noop),
          ])
        );
      },
    });

    result && expect(generate(result).code).toMatchSnapshot();
  });
});
