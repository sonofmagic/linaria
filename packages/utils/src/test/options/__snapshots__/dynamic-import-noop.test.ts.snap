// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic-import-noop case 0 1`] = `
"({
  then: () => undefined,
  catch: () => undefined
});"
`;
